### DISCUSSION TITLE
Any advice if I cannot downgrade bios(uefi) verison?

### DISCUSSION THREAD

**Discussion by CicadaSeventeen** [fwupd/fwupd#5204] [Q&A] [answered]
New bios of `HP EliteBook 845 G9`  breaks Linux. I find older verison from fwupd.org. However, when I tried to flash it in, though `fwupd` did not report any error, I cannot downgrade the version. Because when it rebooted after flashed it would say something like 'bios version is locked' and refused to complete the downgrading.
Is there any way to write it in?

**Comment by hughsie** [comment]
> New bios of HP EliteBook 845 G9 breaks Linux

Did you install the BIOS update using fwupd and the LVFS? Can you share how it breaks Linux please? Then I can tag the right people.

**Reply by CicadaSeventeen** [reply]
No. Bios version in LVFS is good. However, HP provides newer bios version on its sites,  and it breaks Linux with many PCIE errors. However, there is no way to install bios from LVFS since that device has installed newer version bios.

**Reply by CicadaSeventeen** [reply]
The problem is that now Linux is meeting PCIE and USB errors all the time. Like wifi keeps being turned on and of f all the time.
See here https://h30434.www3.hp.com/t5/Notebook-Operating-System-and-Recovery/EliteBook-845-G9-BIOS-Update-Causing-Problems/td-p/8490697

**Reply by superm1** [reply]
Try resetting BIOS default settings and see if it improves things.

**Reply by superm1** [reply]
I suspect it's this:
https://bugzilla.kernel.org/show_bug.cgi?id=216468

**Reply by CicadaSeventeen** [reply]
> I suspect it's this: https://bugzilla.kernel.org/show_bug.cgi?id=216468

Yes it is. Resetting bios makes nothing better, but blacklisting hp_wmi really helps.

**Reply by superm1** [reply]
I looped in one of the guys that I saw working on hp-wmi lately to that issue.  Hopefully he can shed light on it or come up with a solution.
@hughsie if you have any other contacts feel free to notify them too.

**Comment by superm1** [comment]
Here's the kernel fix for this issue:

https://lore.kernel.org/platform-driver-x86/20221028155527.7724-1-jorge.lopez2@hp.com/T/#u

**Comment by superm1** [comment, marked as answer]
Just to close the loop, this was accepted into 6.1-rc5 to fix it: https://github.com/torvalds/linux/commit/1598bfa8e1faa932de42e1ee7628a1c4c4263f0a
